reference docs for CassandraProjection (#112)
akka/akka-projection: docs/src/main/paradox/cassandra.md

The table below shows `akka-projection-cassandra`'s direct dependencies and the second tab shows all libraries it depends on transitively.

@@dependencies{ projectId="akka-projection-cassandra" }

## at-least-once

> The offset can be stored after the envelope has been processed and that gives at-least-once processing semantics.

Reviewer: ```suggestion
The offset is stored after the envelope has been processed and giving at-least-once processing semantics.
```

This means that if the projection is restarted from previously stored offset some elements may be processed more
than once.

Scala
:  @@snip [CassandraProjectionDocExample.scala](/examples/src/test/scala/docs/cassandra/CassandraProjectionDocExample.scala) { #atLeastOnce }

Java
:  @@snip [CassandraProjectionDocExample.java](/examples/src/test/java/jdocs/cassandra/CassandraProjectionDocExample.java) { #atLeastOnce }

The `saveOffsetAfterEnvelopes` and `saveOffsetAfterDuration` parameters control how often the offset is stored.
There is a performance benefit of not storing the offset too often but the drawback is that there can be more
duplicates when the projection is restarted.

The @ref:[`ShoppingCartHandler` is shown below](#handler).

## at-most-once

> The offset for each envelope can be stored before the envelope has been processed and that gives at-most-once

Reviewer: ```suggestion
The offset for each envelope is stored before the envelope has been processed and giving at-most-once
```

processing semantics. This means that if the projection is restarted from previously stored offset one envelope
may not have been processed.

Scala
:  @@snip [CassandraProjectionDocExample.scala](/examples/src/test/scala/docs/cassandra/CassandraProjectionDocExample.scala) { #projection-imports #atLeastOnce }

Java
:  @@snip [CassandraProjectionDocExample.java](/examples/src/test/java/jdocs/cassandra/CassandraProjectionDocExample.java) { #projection-imports #atLeastOnce }

> Since the offset must stored for each envelope this is slower than @ref:[at-least-once](#at-least-once), which

Reviewer: ```suggestion
Since the offset must be stored for each envelope this is slower than @ref:[at-least-once](#at-least-once), which
```
